repository/src/main/java/org/jbpm/process/workitem/repository/service/RepoService.java

```diff
@@ -121,17 +124,59 @@ protected void loadModules() {
         }
     }
     
+    /*
+     * Service related operations with resultMap
+     */
+
+    public void addService(RepoData service, Map<String, List<String>> resultMap) {
+        if (!addNewService(service, resultMap)) {
+            updateService(service, resultMap);
+        }
+
+        listeners.forEach(listener -> listener.onServiceTaskAdded(service));
+    }
+
     /*
      * Service related operations
      */
-    
-    public void addService(RepoData service) {        
+
+    public void addService(RepoData service) {
         services.add(service);
         storage.onAdded(service);
 
         listeners.forEach(listener -> listener.onServiceTaskAdded(service));
     }
 
+    private boolean addNewService(RepoData service, Map<String, List<String>> resultMap) {
+        List<RepoData> addNewRepoData = services.stream().filter(repoData -> repoData.getName().equals(service.getName())).collect(Collectors.toList());
+        if (addNewRepoData.size() < 1 && services.size() > 1) {
```

Comment:
maybe replace this condition with `addNewRepoData.isEmpty() && !services.isEmpty()`

Reply:
done